apps/wpcom-block-editor/src/common/tracking/wpcom-block-editor-edit-template-part.js

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable import/no-extraneous-dependencies */
+/**
+ * External dependencies
+ */
+import { debounce, get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import tracksRecordEvent from './track-record-event';
+
+const trackFullSiteEditingEditTemplatePart = ( event, target ) => {
+	const templatePartType = get( target, [ 'dataset', 'templatePartType' ] );
+	tracksRecordEvent( 'wpcom_block_editor_a8c_fse_edit_template_part', {
+		template_part: templatePartType,
+	} );
+};
+
+/**
+ * Return the event definition object to track `wpcom_block_editor_a8c_fse_edit_template_part`.
+ *
+ * @return {{handler: function, selector: string, type: string}} event object definition.
+ */
+export default () => ( {
+	selector: '.template__block-container .template-block__overlay a',
+	type: 'click',
+	handler: debounce( trackFullSiteEditingEditTemplatePart, 1000, { leading: true } ),
```

Comment:
What happens without the debounce?

Reply:
Ah every render I see. Maybe it's time to turn TemplateEdit into a class? I feel like it's pretty complex for a  component funciton.

Reply:
Basically we `preventDefault` on click in order to save the post first, and trigger a second click to navigate to the template editor.

https://github.com/Automattic/wp-calypso/blob/46879183ea9d0c7f36e9235e151564077ace5df1/apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/edit.js#L126-L133

https://github.com/Automattic/wp-calypso/blob/46879183ea9d0c7f36e9235e151564077ace5df1/apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/template/edit.js#L97-L106

I wonder if it would be better to add `stopPropagation` to the save function instead of debouncing the tracker. 🤔

Reply:
How about explicitly firing from one of the handlers then? Alternatively we can fire when loading the edit single template view (like a page view), unless it's important we catch the click event. (Folks can also visit routes directly)

Reply:
We are avoiding introducing WPCOM-specific code into the plugin.
For example, when we added logging for template data population failures, we did so with an action hook created on WPCOM-side.

I thought of firing on page load, especially because we have the `fse_parent_post` query param that would basically mean folks navigated to the template editor through the normal page editor flow. (Of course they could directly open an URL with `fse_parent_post`, but I guess it might be a minor issue?)
Eventually, when I found this tracking library, I ended up reasoning against creating a new piece of code just for this specific case, and in favour of keeping all the WPCOM block editor tracking in the same place (as much as possible, at least).

Reply:
We can still avoid wpcom specific code. The approach would be similar to https://github.com/Automattic/wp-calypso/pull/35961/files but instead of `a8c_fse_log` we'd have a `a8c_fse_track` action for server events to opt-into

Reply:
But then, AFAIK, we'd need a whole new infrastructure to use a PHP defined hook in JS.
I don't think it's worth it, to be honest 🤔

Reply:
If this is more of a page view, it'd be all server side.

Reply:
Right, that would make sense